### ISSUE TITLE
please release swe-enviroment

### ISSUE THREAD

**Issue by yyht** [zhenyuhe00/SWE-Swiss#1]
Nice work!
Could you release swe-enviroment and rl recipe for online-RL-training? I think this is far more important than sft.

**Comment by zhenyuhe00** [comment]
Thanks for the kind words and the excellent question.

**1. SWE Environment for Online RL**
The core swe-environment is publicly available from [SWE-Smith](https://github.com/SWE-bench/SWE-smith) and [SWE-Gym](https://github.com/SWE-Gym/SWE-Gym). For effective online RL training, the primary engineering endeavor is deploying this Docker-based environment across multiple machines. This is necessary to provide the timely reward feedback required for efficient training.

**2. RL Recipe & Hyperparameters**
Regarding the RL recipe, more efforts are required to align our internal codebase with the public open-source verl for a clean release. However, to aid in reproducibility, we are happy to share the hyperparameters and key techniques from our recipe:

* **Training Configuration:**
    * `prompt_batch_size`: 128
    * `rollouts_per_prompt`: 8
    * `ppo_mini_batch_size`: 128 (This results in 8 gradient updates for each rollout step)
    * `learning_rate`: 1e-6 (constant, with 10 warmup steps)

* **Sequence Lengths:**
    * `max_prompt_length`: 65,536
    * `max_response_length`: 32,768

* **Techniques adopted from DAPO:**
    * Dynamic Batch Size
    * `clip_high`: 0.28
    * No KL Loss (`kl_coef` = 0)
    * Token-level Policy Gradient Loss

We hope this information is helpful。

**Comment by ZhonghaoJiang** [comment]
Are there any plans to release the complete RL training code?
